### ISSUE TITLE
Add album playback mode

### ISSUE THREAD

**Issue by Forage** [MusicPlayerDaemon/MPD#1167]
## Feature request
What I'm missing in MPD is an album shuffle mode. A way to have it randomly pick an album from the playlist, play that album from start to finish, followed by randomly picking a new album to play, etc.

With the existing playback options (consume, random, repeat and single), if album mode would be enabled, either: 
1. all could be functioning on album level; repeat the whole album, stop after having played the whole album, etc
2. have every option work on song level; repeat the one song when repeat was enabled, stop after the one song, play the songs randomly within the random album
3. or random could be the only exception; repeat the one song when repeat was enabled, stop after the one song, etc but have random only act as the indicator for picking a random album

I'd say option one would be the most obvious and predictable to the user and has therefore my preference.

What do you think?

**Comment by MaxKellermann** [comment]
I don't quite understand. You want an additional "album mode", which is a modifier for (some of) the other modes?
Let's start with random mode: you said "only act as the indicator for picking a random album", but under what circumstances shall MPD pick a random album?
This also needs a precise definition what an "album" is.

**Comment by Forage** [comment]
Yes, an additional `album mode` that can be toggled on and off. Based on the functioning of the current playback options (consume, random, repeat and single) this would have an impact on those when this mode is enabled.

Option 1 of implementing it:
To make it do what I'm after (album shuffle) I would enable `album mode` and `random` and start playing. It will pick a random album, play the first track until the last of that album, and pick a different album randomly again. If I don't have `random` enabled it will simply play the next album in the playlist at the end. Unless e.g. `repeat` is enabled, in which case it would repeat the current album from start to finish again.

Option 2 of implementing it:
Enable `album mode` and start playing. `album mode` in this case would imply a random on album level. It will pick a random album, play the first track until the last of that album, and pick a different album randomly again. When I activate `random` on top of it it will not play the picked album from the first to the last track, but play each track within the album in a random order, followed by randomly picking a different album again. Enable `repeat` and only the song that was playing when it was enabled is going to be repeated.

Option 3 of implementing it:
Like option 2, but in this case `album mode` does not imply a random on album level unless you enable `random` as well.

The definition of "album" depends on the presence of the 'album name' and 'album artist' tags.
Songs without an 'album name' tag are either ignored of considered an album all together.
Songs the same 'album name' tag are considered the same album, but separated by 'album artist' tag if present.

Playing from start to finish means lowest to highest 'track number', because tracks could be missing.

**Comment by mxmilkiib** [comment]
There are a [few](https://github.com/axujen/mpdrandom) [scripts](https://github.com/f0rki/mpd-album-shuffle) [etc.](https://bitbucket.org/tim_heap/mpd-random-album/src/master/) [that](https://github.com/joshkunz/ashuffle) [do](https://github.com/ibeex/mpd_queue_random_album) random album shuffle, but these require command-line access. The ability to natively (in clients) enable jumping to the start of a random album when the playing album finishes would be amazing.

**Comment by jcorporation** [comment]
The myMPD jukebox mode does exactly this: https://github.com/jcorporation/myMPD

**Comment by aqxa1** [comment]
`ncmpcpp` commandline client has _kind of_ supported this at a basic level for years, although it instead populates a playlist with `n` number of random albums/artists/songs etc. But there is no awareness that it is storing albums so you can't skip/forward or back at the album level.

So agreed, I'd love to see this feature happen.
